mycore-viewer/src/main/typescript/modules/iiif/widgets/IIIFStructureBuilder.ts

```diff
@@ -78,28 +80,45 @@ namespace mycore.viewer.widgets.iiif {
             return this.vStructureModel;
         }
 
-        private processChapter(parent: model.StructureChapter, chapter: IRange): model.StructureChapter {
+        private processChapter(parent: model.StructureChapter, chapter: IRange, cans: ICanvas[]): model.StructureChapter {
             // if (chapter.nodeName.toString() == "mets:mptr") {
             //     return;
             // }
             //TODO Chaptertype currently not in Manifest
-            const chapterObject = new model.StructureChapter(parent,
-                '', this.getIDFromURL(chapter.id), chapter.getDefaultLabel());
-            // let chapterChildren = chapter.getRanges();
-
-            this.vChapterIdMap.set(chapterObject.id, chapterObject);
+            let chapterObject;
+            if (chapter === undefined || (chapter.getCanvasIds().length === 0 && chapter.getRanges().length === 0)) {
+                chapterObject = new model.StructureChapter(parent,
+                    '', 'LOG_0', this.manifestDocument.getDefaultLabel());
+                this.vChapterIdMap.set(chapterObject.id, chapterObject);
+                cans.forEach((can: ICanvas) => {
+                    const childChap = new model.StructureChapter(chapterObject,
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['model' is not defined. (no-undef)](https://app.codacy.com/app/MyCoRe/mycore/pullRequest?prid=4613896)